packages/core/mesh/network-manager/src/swarm/connection.ts

```diff
@@ -190,6 +198,41 @@ export class Connection {
     log.trace('dxos.mesh.connection.open-connection', trace.end({ id: this._instanceId }));
   }
 
+  @synchronized
+  // TODO(nf): make the caller responsible for recording the reason and determining flow control.
+  async abort(err?: Error) {
+    log('abort');
```

Comment:
we generally do
`log('aborting...', { err })`
for things that might take time or fail.
you also have `aborting...` below.

Reply:
makes sense. was mostly just getting some early feedback on the overall approach here, and roughly had added the debugging traces that i thought might be useful later on. but can see how attention to detail in these complicated and progressive flows makes it easier to reason about.